### ISSUE TITLE
Factor out chunked reader from Client::request_events

### ISSUE THREAD

**Issue by clux** [clux/kube-rs#379]
Currently the chunk buffering is inlined into the `request_events` call.
This makes the function huge, hard to follow, and hard to test.

It would be better if there was a more cleanly separated `Reader` like struct that dealt with buffering and error handling of un-closed chunks specifically.

Not a big issue, but it makes that code a bit awkward.

**Comment by kazk** [comment]
Yeah, I was going to ask about this. Is the response stream of line delimited JSON? Can we use `tokio_util::codec::FramedRead` with `tokio_util::codec::LinesCodec`?

**Comment by clux** [comment]
Ooh, yeah those look promising. The body of the fn looks very much like an impl Decoder at the very least.

Yes, it's newline delimited JSON. You have to test it after each chunk to see if you have a complete object since the objects can exceed MTU.
